packages/template-retail-react-app/app/hooks/use-is-hydrated.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useState, useEffect} from 'react'
+
+/**
+ * Hook to determine if the code is hydrated
+ * Solution for hydration issues with SSR.
+ * @returns {boolean} true if the code is finished hydration
+ */
+export const useIsHydrated = () => {
```

Comment:
Have you tried `window.__HYDRATING__`? 

see https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx#L122

Reply:
oh ignore me, i just saw that you changed the SDK.